fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java

```diff
@@ -183,6 +192,116 @@ public DatatableData retrieveDatatable(final String datatable) {
         return datatableData;
     }
 
+    @Override
+    public List<JsonObject> queryDataTable(String datatable, String columnFilter, String valueFilter, String resultColumns) {
+        Arrays.asList(datatable, columnFilter, valueFilter, resultColumns).forEach(SQLInjectionValidator::validateDynamicQuery);
+        List<String> resultColumnNames = Stream.of(resultColumns.split(",")).toList();
+        String sql = "select " + resultColumns + " from " + datatable + " where " + columnFilter + " = ?";
+
+        List<ResultsetColumnHeaderData> resultsetColumnHeaderData = genericDataService.fillResultsetColumnHeaders(datatable);
+        Object finalValueFilter = valueFilter;
+        SqlRowSet rowSet = null;
+        String filterColumnType = resultsetColumnHeaderData.stream().filter(column -> columnFilter.equals(column.getColumnName()))
+                .findFirst().map(ResultsetColumnHeaderData::getColumnType).orElse(columnFilter + " does not exist in datatable");
+        if (databaseTypeResolver.isPostgreSQL()) {
+            int[] argType = new int[1];
+            if ("bit".equalsIgnoreCase(filterColumnType)) {
+                finalValueFilter = BooleanUtils.toString(BooleanUtils.toBooleanObject(valueFilter), "1", "0", "null");
+                argType[0] = Types.BIT;
+            } else if ("boolean".equalsIgnoreCase(filterColumnType) || "bool".equalsIgnoreCase(filterColumnType)) {
+                finalValueFilter = BooleanUtils.toString(BooleanUtils.toBooleanObject(valueFilter), "true", "false", "null");
+                argType[0] = Types.BOOLEAN;
+            } else if ("integer".equalsIgnoreCase(filterColumnType)) {
+                argType[0] = Types.INTEGER;
+            } else if ("bigint".equalsIgnoreCase(filterColumnType)) {
+                argType[0] = Types.BIGINT;
+            } else if ("date".equalsIgnoreCase(filterColumnType)) {
+                argType[0] = Types.DATE;
+            } else if (filterColumnType.toLowerCase().contains("timestamp")) {
+                argType[0] = Types.TIMESTAMP;
+            } else if ("numeric".equalsIgnoreCase(filterColumnType)) {
+                argType[0] = Types.DECIMAL;
+            } else if ("text".equalsIgnoreCase(filterColumnType) || "character varying".equalsIgnoreCase(filterColumnType)) {
+                argType[0] = Types.VARCHAR;
+            } else {
+                List<ApiParameterError> paramErrors = new ArrayList<>();
+                paramErrors.add(parameterErrorWithValue("400",
+                        "Unsupported input type for datatable query! Column filter: " + filterColumnType, "valueFilter", valueFilter));
+                throw new PlatformApiDataValidationException(paramErrors);
+            }
+            rowSet = jdbcTemplate.queryForRowSet(sql, new Object[] { finalValueFilter }, argType);
```

Comment:
I don't think we need the `new Object[]`

Reply:
I used the same syntax as before. There are numerous cases around the project.

Reply:
Got it, but still this is not needed. As a matter of fact we started to get rid of those slowly but surely.

Reply:
the problem with this is that the queryForRowSet method has 2 different signatures:
public SqlRowSet queryForRowSet(String sql, Object[] args, int[] argTypes) 
public SqlRowSet queryForRowSet(String sql, @Nullable Object... args)
if I omit the object array, it will resolve to the latter and it won't take the argTypes correctly